mobile/src/feeds/home-feeds-screen.tsx

```diff
@@ -1,28 +1,87 @@
 import React from 'react';
-import { FlatList, Text, View} from 'react-native';
+import { FlatList, Text, View, StyleSheet, ListRenderItemInfo, RefreshControl } from 'react-native';
 import { TopNavBar } from '../topNavBar/top-nav-bar';
 
-const navBar =  new TopNavBar('Feeds');
-const dataList = [
-{key: 'First Feed'},
-{key: 'Second Feed'},
-{key: 'Third Feed'},
-{key: 'Fourth Feed'},
-{key: 'Fifth Feed'},
-{key: 'Sixth Feed'},
-{key: 'Seventh Feed'},
-{key: 'Ten'}
-];
+import { Employee } from '../reducers/organization/employee.model';
+import { EmployeesStore } from '../reducers/organization/employees.reducer';
+import { Feed } from '../reducers/feeds/feed.model';
+import { connect } from 'react-redux';
+import { AppState } from '../reducers/app.reducer';
 
+import { FeedListItem } from './feed';
 
-export class HomeFeedsScreen extends React.Component {
+import { screenStyles as styles } from './styles';
+
+const navBar = new TopNavBar('Feeds');
+
+interface FeedsScreenProps {
+    feeds: Feed[];
+    employees: EmployeesStore;
+}
+
+interface FeedsScreenState {
+    employees?: Employee[];
```

Comment:
Please explain why is it better than having `employeesById: state.organization.employees.employeesById` in props?
The way I see it, `render` will be called anyway if nextProps is different (and it will be different), but there is additional setState and map function in addition (loop + array allocation).

Reply:
Aaaand you r wrong. FlatList won't rerender unless feeds list change. That is why we need `employess` list that we assign lower to the `extraData` property of a list